repo-settings.c

```diff
@@ -36,6 +37,30 @@ void prepare_repo_settings(struct repository *r)
 	/* Defaults modified by feature.* */
 	if (experimental) {
 		r->settings.fetch_negotiation_algorithm = FETCH_NEGOTIATION_SKIPPING;
+
+		/*
+		 * Force enable the builtin FSMonitor (unless the repo
+		 * is incompatible or they've already selected it or
+		 * the hook version).  But only if they haven't
+		 * explicitly turned it off -- so only if our config
+		 * value is UNSET.
+		 *
+		 * lookup_fsmonitor_settings() and check_for_ipc() do
+		 * not distinguish between explicitly set FALSE and
+		 * UNSET, so we re-test for an UNSET config key here.
+		 *
+		 * I'm not sure I want to fix fsmonitor-settings.c to
+		 * have more than one _DISABLED state since our usage
+		 * here is only to support this experimental period
+		 * (and I don't want to overload the _reason field
+		 * because it describes incompabilities).
+		 */
+		if (manyfiles &&
+		    fsmonitor_ipc__is_supported()  &&
+		    fsm_settings__get_mode(r) == FSMONITOR_MODE_DISABLED &&
+		    (repo_config_get_maybe_bool(r, "core.fsmonitor", &value) > 0 ||
+		     repo_config_get_bool(r, "core.useBuiltinFSMonitor", &value)))
```

Comment:
If I read this correctly, FSMonitor is now enabled upon `experimental.manyFile` unless _both_ `core.fsmonitor` _and_ `core.useBuiltinFSMonitor` are configured (and it overrides `core.fsmonitor=<path>`). I think it should read like this instead:

```c
		if (manyfiles &&
		    fsmonitor_ipc__is_supported()  &&
		    fsm_settings__get_mode(r) == FSMONITOR_MODE_DISABLED &&
		    repo_config_get_maybe_bool(r, "core.fsmonitor", &value) ==1 /* unset */ &&
		    repo_config_get_bool(r, "core.useBuiltinFSMonitor", &value) /* also unset */)
```

Reply:
Fixed, now in commit 3dd2f446b5